### ISSUE TITLE
All requiring jobs start with "pr" are expanded like pull request job

### ISSUE THREAD

**Issue by yoshwata** [screwdriver-cd/screwdriver#1417]
**What happened**:
Set `~promote` as `requres`, then it is expanded like `~pr`.

**What you expected to happen**:
The job set `~promote` as `requires` is expanded to be executed after `promote` job.
That is not pull request job.

**How to reproduce it**:
Set the yaml like below to the validator 
```
shared:
  image: node8:latest
jobs:
  main:
    requires: [ ~pr, ~commit ]
    steps:
      - echo
  promote:
    requires: [ ~main ]
    steps:
      - echo
  clean:
    requires: [ ~promote ]
    steps:
      - echo
```

then the result is
<img width="203" alt="2018-12-17 15 35 20" src="https://user-images.githubusercontent.com/32473622/50070468-6db83880-0211-11e9-809c-7c16a72ec8bb.png">

`clean` job should be executed after promote.

**Comment by yokawara** [comment]
Jobs which names start with `commit` also cause the same problem.

**Comment by tkyi** [comment]
Is it because of code with `startsWith` comparison?
Like: https://github.com/screwdriver-cd/workflow-parser/blob/master/lib/getNextJobs.js#L43

**Comment by tkyi** [comment]
Should be resolved with `API: v0.5.782`

<img width="1171" alt="Screen Shot 2019-11-04 at 4 48 56 PM" src="https://user-images.githubusercontent.com/3230529/68169999-038cad00-ff23-11e9-84e2-085fa0ac5cfe.png">
